online-shop/src/app/product.service.ts

```diff
@@ -0,0 +1,42 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from "@angular/common/http";
+import {find, Observable} from "rxjs";
+import {Product} from "../product";
+import {ShoppingCart} from "../shopping-cart";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductService {
```

Comment:
Here are the SOLID principles: https://www.freecodecamp.org/news/solid-principles-explained-in-plain-english/.
this service should handle just the httpRequests and not the logic of the cart(addNewItem, increment if exists and so on)
As an idea you can have a CartService that you inject in the app.component so it will maintain the status along the whole session.
Here you will handle the addToCart logic, increment value, remove from cart and so on. and after all modifications are applied then you call the ProductService to make the httpRequest.